### PR TITLE
add automatically_add_executor_with_node option

### PR DESCRIPTION
**Pull request by BriceRenaudeau**
This PR tries to implement #1589

It adds the `automatically_add_executor_with_node` option when creating a callback_group for a LifecycleNode.

### CODE REVIEW

File: rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_node.hpp [hunk 1]
```diff
@@ -183,7 +183,9 @@ class LifecycleNode : public node_interfaces::LifecycleNodeInterface,
    */
   RCLCPP_LIFECYCLE_PUBLIC
   rclcpp::CallbackGroup::SharedPtr
-  create_callback_group(rclcpp::CallbackGroupType group_type);
+  create_callback_group(
+    rclcpp::CallbackGroupType group_type,
+    bool automatically_add_executor_with_node = true);
```
Comment (fujitatomoya) [inline rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_node.hpp:188]:
```suggestion
    bool automatically_add_to_executor_with_node = true);
```

how about using the same name with
https://github.com/ros2/rclcpp/blob/3fa511a8a03682eb5e876c0def204b79bd3491c4/rclcpp/include/rclcpp/node.hpp#L146

**Comment by fujitatomoya** [comment]
@BriceRenaudeau 

thanks for opening PR :+1: 

could you also take care of DCO failure https://github.com/ros2/rclcpp/pull/1594/checks?check_run_id=2183190459?

File: rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_node.hpp (hunk 1 above)
Comment (BriceRenaudeau) [reply rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_node.hpp:188]: You are right its a typo.

**Review by wjwwood — APPROVED** [review]

File: rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_node.hpp [hunk 2]
```diff
@@ -183,7 +183,9 @@ class LifecycleNode : public node_interfaces::LifecycleNodeInterface,
    */
```
Comment (wjwwood) [inline rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_node.hpp:183]: Actually, can you document this new parameter?

**Review by wjwwood — CHANGES_REQUESTED** [review]
Also, since this API is not covered by https://github.com/ros2/rclcpp/blob/master/rclcpp/test/rclcpp/test_add_callback_groups_to_executor.cpp, can you maybe add some code here to just exercise the new API?

https://github.com/ros2/rclcpp/blob/0088e35052270eca4ccbb0f0c2325f4030c39baf/rclcpp_lifecycle/test/test_lifecycle_node.cpp#L663-L675

**Review by wjwwood — APPROVED** [review]

**Comment by wjwwood** [comment]
@ros-pull-request-builder retest this please

**Comment by wjwwood** [comment]
CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14085/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8903/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11778/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14179/)

**Pull request merged**